mopidy_musicbox_webclient/static/js/functionsvars.js

```diff
@@ -176,185 +176,156 @@ function albumTracksToTable (pl, target, uri) {
-        name = song.name
+        iconClass = 'trackname'
     }
-    songLi = '<li class="song albumli" id="' + liID + '">' +
-             '<a href="#" class="moreBtn" onclick="return popupTracks(event, \'' + uri + '\',\'' + song.uri + '\');">' +
+    // Play by tlid if available.
+    if (tlid) {
+        tlidString = '" tlid="' + tlid
+        tlidParameter = '\',\'' + tlid
+        onClick = 'return playTrackQueueByTlid(\'' + song.uri + '\',\'' + tlid + '\');'
+    } else {
+        onClick = 'return playTrackByUri(\'' + song.uri + '\',\'' + uri + '\');'
+    }
+    songLi = '<li class="song albumli" id="' + liID + tlidString + '">' +
+             '<a href="#" class="moreBtn" onclick="return popupTracks(event, \'' + uri + '\',\'' + song.uri + tlidParameter + '\');">' +
              '<i class="fa fa-ellipsis-v"></i></a>' +
-             '<a href="#" onclick="return playTrackByUri(\'' + song.uri + '\',\'' + uri + '\');">' +
-             '<h1 class="trackname">' + name + '</h1></a>' +
-             '</li>'
+             '<a href="#" onclick="' + onClick + '">' +
+             '<h1><i class="' + iconClass + '"></i>' + song.name + '</h1>'
+    if (renderAlbumInfo) {
+        songLi += '</p>'
+        songLi += renderSongLiTrackArtists(song)
+        if (song.album && song.album.name) {
+            songLi += ' / '
+            songLi += '<em>' + song.album.name + '</em></p>'
+        }
+    }
+    songLi += '</a></li>'
     return songLi
 }
 
-function renderQueueSongLi (song, liID, uri, tlid) {
-    var name
-    if (!song.name || song.name === '') {
-        name = uri.split('/')
-        name = decodeURI(name[name.length - 1])
+function renderSongLiTrackArtists (track) {
+    var html = ''
+    if (track.artists) {
+        for (var i = 0; i < track.artists.length; i++) {
+            html += track.artists[i].name
+            html += (i === track.artists.length - 1) ? '' : ' / '
+            // Stop after 3
+            if (i > 2) {
+                html += '...'
+                break
+            }
+        }
+    }
+    return html
+}
+
+function isNewAlbumSection (track, previousTrack) {
+    // 'true' if album name is either not defined or has changed from the previous track.
+    return !track.album || !track.album.name || !previousTrack || !previousTrack.album || !previousTrack.album.name ||
+           (previousTrack && previousTrack.album && previousTrack.album.name && track.album.name !== previousTrack.album.name)
+}
+
+function isMultiTrackAlbum (track, nextTrack) {
+    // 'true' if there are more tracks of the same album after this one.
+    return nextTrack.album && nextTrack.album.name && track.album && track.album.name && track.album.name === nextTrack.album.name
+}
+
+function validateTrackName (track) {
+    // Create name if there is none
+    var name = ''
+    if (!track.name || track.name === '') {
+        name = track.uri.split('/')
+        name = decodeURI(name[name.length - 1]) || 'Track ' + String(i)
```

Comment:
``i`` came out of a ``for`` loop - I missed this when I moved this to the helper function.

I'll just pass the track number in to the helper function so that we can retain the feature to rename unnamed tracks to 'Track n'.